src/mots/settings.py

```diff
@@ -0,0 +1,73 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Set up the mots environment.
+
+This module fetches various variables either from the environment or config files that
+will directly affect the operation of mots after startup.
+
+Variables defined in this module may be overridden by environment variables or
+in the overrides file.
+"""
+
+import os
+from pathlib import Path
+
+from mots import __version__
+from mots.yaml import yaml
+
+HOME_DIRECTORY = Path.home()
+RESOURCE_DIRECTORY = HOME_DIRECTORY / ".mots"
+OVERRIDES_FILE = RESOURCE_DIRECTORY / "settings.yaml"
+
+
+class Settings:
+    """Helper class to convert settings dict to object."""
+
+    DEFAULTS = {
```

Comment:
I was thinking a bit more about this last night and I'm wondering if this class would make sense as a dataclass. Most of the types are scalars here so we would be able to define defaults like `LOG_BACKUPS: int = 5`. This would enable removing the `setattr` method of defining properties which in my experience can balloon into issues that are quite tough to track down. It would also enable adding type hints to all the values.

I won't block the review on it but I wanted to throw it out there for your consideration.

Reply:
Dataclasses were considered however we are targeting Python 3.6 which does not have dataclasses and I do not want to add any backports or additional dependencies.

This class is well tested so I don't foresee any issues emerging here (though a couple more tests re: type casting could be added).